### PR TITLE
tests: email/password register user

### LINKED ISSUE #453: Implement tests: Auth Provider: register email

### PR DESCRIPTION
**Pull request by desistefanova**
Testing register user with email/password authentication provider. Fixes #453

### CODE REVIEW

File: test/credentials_test.dart [hunk 1]
```diff
@@ -36,4 +36,61 @@ Future<void> main([List<String>? args]) async {
...
+  baasTest('Email/Password - register user', (configuration) async {
+    final app = App(configuration);
+    final authProvider = EmailPasswordAuthProvider(app);
+    String emailPrefix = generateRandomString(5);
+    String username = "realm_tests_do_autoverify$emailPrefix@bar.com";
+    String password = "SWV23R#@T#VFQDV";
+    await authProvider.registerUser(username, password);
+    final user = await app.logIn(Credentials.emailPassword(username, password));
+    await app.logout(user);
+    await app.removeUser(user);
```
Comment (desistefanova) [inline test/credentials_test.dart:60]: By some reason the user is not removed from the app. Should be investigated.

**Review by nielsenko — APPROVED** [review]

File: test/credentials_test.dart (hunk 1 above)
Comment (nirinchev) [reply test/credentials_test.dart:60]: We probably don't these here and we'll write separate tests for user.logout/app.removeUser.

Comment (nirinchev) [reply test/credentials_test.dart:60]: We probably don't these here and we'll write separate tests for user.logout/app.removeUser.

**Review by nirinchev — APPROVED** [review]

File: test/credentials_test.dart (hunk 1 above)
Comment (desistefanova) [reply test/credentials_test.dart:60]: Yes sure.

**Review by nirinchev — CHANGES_REQUESTED** [review]
These tests appear not to be running on CI. Can you hold off on merging them until we figure out what's going on? They are skipped but no reason is provided:

Screen Shot 2022-04-29 at 01 26 19

**Comment by desistefanova** [comment]
> These tests appear not to be running on CI. Can you hold off on merging them until we figure out what's going on? They are skipped but no reason is provided:
> 
> Screen Shot 2022-04-29 at 01 26 19

I think the problem is in this row in `baasTest`
```
if (skip == null) {
    skip = url == null ? "BAAS URL not present" : true;
```
I will try to fix it.

**Comment by nirinchev** [comment]
It looks like there's a bug in the skip logic: https://github.com/realm/realm-dart/blob/f3920a70bf48d82536225f2a1e352a7191d9e563/test/test.dart#L221-L223

The way I read it, if the url is null, the test will be skipped with reason `BAAS URL not present`, otherwise it will just be skipped because `skip` is set to `true`.

**Comment by desistefanova** [comment]
> Edit: seems like you were faster than me 😅
> 
> It looks like there's a bug in the skip logic:
> 
> https://github.com/realm/realm-dart/blob/f3920a70bf48d82536225f2a1e352a7191d9e563/test/test.dart#L221-L223
> 
> The way I read it, if the url is null, the test will be skipped with reason `BAAS URL not present`, otherwise it will just be skipped because `skip` is set to `true`.

Yes, I was looking at this just right now.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -277,6 +277,41 @@ jobs:
         working-directory: ./flutter/realm_flutter/tests
 
 # Windows jobs
+  baas-windows:
+    runs-on: ubuntu-latest
```
Comment (blagoev) [inline .github/workflows/ci.yml:281]: this is seems strange. is this intentional

File: test/test.dart [hunk 3]
```diff
@@ -88,9 +88,9 @@ class $RemappedClass {
 }
 
 String? testName;
-Map<String, BaasApp> baasApps = <String, BaasApp>{};
-
+final baasApps = <String, BaasApp>{};
 final _openRealms = Queue<Realm>();
+late Uri? url;
```
Comment (blagoev) [inline test/test.dart:93]: why a global var

**Review by blagoev — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (nirinchev) [reply .github/workflows/ci.yml:281]: What is strange - the fact we're adding baas tests for Windows or that the deploy step runs on ubuntu?

File: test/baas_test.dart [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import 'credentials_test.dart' as credentials_tests;
+import 'app_test.dart' as app_tests;
+
+Future<void> main([List<String>? args]) async {
```
Comment (nirinchev) [inline test/baas_test.dart:4]: Do we need this file?

File: test/test.dart (hunk 3 above)
Comment (desistefanova) [reply test/test.dart:93]: This will be removed and I will take the code from master.

File: test/baas_test.dart (hunk 4 above)
Comment (desistefanova) [reply test/baas_test.dart:4]: No more

Comment (desistefanova) [reply test/baas_test.dart:4]: We don't need it anymore.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (blagoev) [reply .github/workflows/ci.yml:281]: I misread this. Closing

**Review by blagoev — CHANGES_REQUESTED** [review]

**Pull request merged**